homeassistant/components/vicare/climate.py

```diff
@@ -0,0 +1,363 @@
+        ]
+    )
+
+
+class ViCareClimate(ClimateDevice):
+    """Representation of the ViCare heating climate device."""
+
+    def __init__(self, hass, name, api):
+        """Initialize the climate device."""
+        self._name = name
+        self._state = None
+        self._api = api
+        self._support_flags = SUPPORT_FLAGS_HEATING
+        self._unit_of_measurement = hass.config.units.temperature_unit
+        self._target_temperature = None
+        self._current_mode = VALUE_UNKNOWN
+        self._current_temperature = None
+        self._current_program = VALUE_UNKNOWN
+
+    def update(self):
+        """Let HA know there has been an update from the ViCare API."""
+        _room_temperature = self._api.getRoomTemperature()
+        if _room_temperature is not None and _room_temperature != "error":
+            self._current_temperature = _room_temperature
+        else:
+            self._current_temperature = self._api.getBoilerTemperature()
+        self._current_program = self._api.getActiveProgram()
+
+        # The getCurrentDesiredTemperature call can yield 'error' (str) when the system is in standby
+        desired_temperature = self._api.getCurrentDesiredTemperature()
+        if desired_temperature == PYVICARE_ERROR:
+            desired_temperature = None
+
+        self._target_temperature = desired_temperature
+
+        self._current_mode = self._api.getActiveMode()
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return self._support_flags
+
+    @property
+    def name(self):
+        """Return the name of the climate device."""
+        return self._name
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return self._unit_of_measurement
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._current_temperature
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._target_temperature
+
+    @property
+    def hvac_mode(self):
+        """Return current hvac mode."""
+        return VICARE_TO_HA_HVAC_HEATING.get(self._current_mode, STATE_UNKNOWN)
+
+    def set_hvac_mode(self, hvac_mode):
+        """Set a new hvac mode on the ViCare API."""
+        _LOGGER.info(
+            "setting hvac mode to %s (%s)",
+            hvac_mode,
+            HA_TO_VICARE_HVAC_HEATING.get(hvac_mode, STATE_UNKNOWN),
+        )
+        self._api.setMode(HA_TO_VICARE_HVAC_HEATING.get(hvac_mode, STATE_UNKNOWN))
+        self.async_schedule_update_ha_state(True)
+
+    @property
+    def hvac_modes(self):
+        """Return the list of available hvac modes."""
+        return list(HA_TO_VICARE_HVAC_HEATING.keys())
+
+    @property
+    def min_temp(self):
+        """Return the minimum temperature."""
+        return convert_temperature(
```

Comment:
We don't need to convert temperatures. It's handled by the base climate class. Just report the temperature in the correct unit.